### PR TITLE
Cml cloud runner

### PR DESCRIPTION
**Pull request by DavidGOrtega**
```yaml
name: train-my-model

on: [push]

jobs:
  deploy:
    runs-on: [ubuntu-latest]
    container: docker://dvcorg/cml:latest

    steps:
      - name: deploy
        shell: bash
        env:
          repo_token: ${{ secrets.REPO_TOKEN }} 
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
        run: |
          echo "Deploying..."

          RUNNER_LABELS="cml"
          RUNNER_REPO="https://github.com/${GITHUB_REPOSITORY}"

          test-cml-cloud-runner \
            --instance_ami ami-03ba3948f6c37a4b0 \
            --instance_type t2.micro \
            --region us-west-1 \
            --repo_token ${repo_token} \
            --runner_repo ${RUNNER_REPO} \
            --runner_labels ${RUNNER_LABELS} \
            --runner_idle_timeout=120 \
            --gpus all

  train:
    needs: deploy
    runs-on: [self-hosted,cml]

    steps:
      - uses: actions/checkout@v2

      - name: cml_run
        env:
          repo_token: ${{ secrets.GITHUB_TOKEN }} 
        run: |
          echo "Training..."
          echo 'Hi from cml!' >> report.md
          # nvidia-smi >> report.md
          cml-send-comment report.md

          echo "Trained!"
```

Additionally the config accepts a terraform file. If specified ignores the instance parameters

```sh
 test-cml-cloud-runner \
            --tf_file "myfile.tf"
            --repo_token ${repo_token} \
            --runner_repo ${RUNNER_REPO} \
            --runner_labels ${RUNNER_LABELS} \
            --runner_idle_timeout=120
```

```sh
Usage: cml-cloud-runner.js

Options:
  --version              Show version number                      [boolean]
  --gpus                 leave empty if no gpu. possible values: all, 1,
                         2... more information read docker gpus param
  -h                     Show help                                [boolean]
  --repo_token                                                   [required]
  --runner_repo                                                  [required]
  --runner_labels                                                [required]
  --runner_name
  --runner_idle_timeout
  --cml_image
  --region
  --instance_ami
  --instance_type
  --tf_file
```

### CODE REVIEW

**Review by dmpetrov — APPROVED** [review]
👍 
I assume it is not the final change.

**Comment by DavidGOrtega** [comment]
TODO: 
 - check URL, token repo and tags before deploy

File: assets/demo.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+terraform {
+  required_providers {
+    iterative = {
+      versions = ["0.1"]
+      source = "github.com/iterative/iterative"
```
Comment (dmpetrov) [inline assets/demo.tf:5]: Do we have this repo\url?

**Review by dmpetrov — APPROVED** [review]
👍  that's a great step to TF

File: assets/demo.tf (hunk 1 above)
Comment (DavidGOrtega) [reply assets/demo.tf:5]: no its inherited from the hashicorp tutorial that sets that bad. I will fix it

Comment (DavidGOrtega) [reply assets/demo.tf]: its a terraform thingy. After fixing it

image

**Pull request merged**